### ISSUE TITLE
Metrics are missing for 5 minutes without any explanation

### ISSUE THREAD

**Issue by karabadzhiys** [wavefrontHQ/wavefront-sdk-java#206]
Pull request [168](https://github.com/wavefrontHQ/wavefront-sdk-java/pull/168) introduces a new logging mechanism that hides connectivity errors for 5 minutes. This introduces undetected 5 minutes delay in alarms that are responsible for autoscaling services, mitigate DDoS stacks, etc. 

Could you make a mechanism to configure the suppressed logger, so the customers can change the suppress interval or entirely disable the suppression. This feature will be very helpful for services that have log analyzers enabled for automatic mitigation of problems when an error/exception is detected in the log (like cutting tickets, restarting the service, etc.).

**Comment by oppegard** [comment]
@karabadzhiys this seems like a reasonable request. My preference is also to change the behavior so that the first log isn't suppressed when an error occurs. For example, if the Wavefront endpoint was offline you'd immediately see:

```
Jan 19, 2023 11:02:10 AM com.wavefront.sdk.common.clients.service.ReportingService safeGetResponseCodeAndClose
SEVERE: Unable to obtain status code from the Wavefront service at https://127.0.0.1:8080/report?f=wavefront due to: java.net.ConnectException: Connection refused (Connection refused)

< 5 minutes later>

Jan 19, 2023 11:07:10 AM com.wavefront.sdk.common.clients.service.ReportingService safeGetResponseCodeAndClose
SEVERE: Unable to obtain status code from the Wavefront service at https://127.0.0.1:8080/report?f=wavefront due to: java.net.ConnectException: Connection refused (Connection refused)

```

**Comment by karabadzhiys** [comment]
There is a benefit of suppressing the logs - if you have an environment (like dev/test) with unstable connectivity to Wavefront. On those environments you may choose to suppress, since most probably the metrics will be sent on the next loop/retry. However IMHO for productive environments the suppression should be disabled.

**Comment by oppegard** [comment]
@karabadzhiys would it be acceptable to change the behavior to log once immediately, before suppressing for 5 minutes? That change could be done quickly and a new version of the SDK released. Adding a new configuration option that is cohesive within the use of different loggers within `WavefrontClient` and `ReportingService` will take more time and thought.

**Comment by karabadzhiys** [comment]
@oppegard Yes that is ok for dev environments to log and suppress for 5 min. However would it be possible to entirely disable the suppression for productive environments?

**Comment by oppegard** [comment]
@karabadzhiys we'll look at one of:

- make the time suppression configurable to override the default of 5 minutes
- make it a boolean of using the suppressing logger vs using a non-supressing logger

**Comment by oppegard** [comment]
@karabadzhiys we've released v3.1.0 of wavefront-sdk-java, which allows you to change the logger's suppression time from the default of 5 minutes:

`WavefrontClient.Builder#reportingServiceLogSuppressTimeSeconds()`
